### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by ieatkillerbees**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
         "laravel/lumen-framework": "5.5.*",
         "ramsey/uuid": "^3.7",
         "ramsey/uuid-doctrine": "^1.4",
-        "vlucas/phpdotenv": "~2.2"
+        "vlucas/phpdotenv": "~2.2",
+        "league/fractal": "^0.17.0"
```
Comment (greydnls) [inline composer.json:17]: did you add this via CLI Or manually?

File: src/Entities/Family.php [hunk 2]
```diff
@@ -113,9 +117,9 @@ public function getPaymentStrategy(): string
     }
 
     /**
-     * @return int
+     * @return int|null
      */
-    public function getCompletionThreshold(): int
+    public function getCompletionThreshold()
```
Comment (greydnls) [inline src/Entities/Family.php:122]: if we tag to 7.1 we can use nullable return types. ` (): ?int`

File: src/Entities/Transformers/Family.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+use League\Fractal\TransformerAbstract;
+
+class Family extends TransformerAbstract
+{
+    const DAYS_OF_WEEK = ['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday'];
+
+    public function transform(Entities\Family $family)
+    {
+        return [
+            'type'       => 'family',
```
Comment (greydnls) [inline src/Entities/Transformers/Family.php:16]: If you use the JsonApi serializer, this isn't necessary. It'll pull the type, ID and links for you automagically. you just return an array.

File: src/Http/Controllers/Family.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+use League\Fractal\Resource\Item;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+class Family extends Controller
+{
+    /**
+     * @var Manager
+     */
+    private $viewManager;
```
Comment (greydnls) [inline src/Http/Controllers/Family.php:16]: `$viewManager` feels like a slightly misleading variable name.

File: src/Entities/Transformers/Family.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace Choredo\Entities\Transformers;
+
+use Assert\Assertion;
+use Choredo\Entities;
+use League\Fractal\TransformerAbstract;
+
+class Family extends TransformerAbstract
+{
+    const DAYS_OF_WEEK = ['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday'];
```
Comment (greydnls) [inline src/Entities/Transformers/Family.php:11]: This already exists here: https://github.com/GreyBee/choredo-api/blob/master/src/Entities/Schedule.php#L7

I'd rather use that constant than repeat it. What do you think?

**Review by greydnls — CHANGES_REQUESTED** [review]

File: src/Entities/Transformers/Family.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Choredo\Entities\Transformers;
```
Comment (greydnls) [inline src/Entities/Transformers/Family.php:3]: IMO Transformers shouldn't be in the Entities Directory. It's not a sub-concern of the entity itself. I'd prefer `Choredo\Transformers` What are your thoughts?

File: src/Http/Controllers/Family.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace Choredo\Http\Controllers;
+
+use Choredo\Entities;
+use League\Fractal\Manager;
+use League\Fractal\Resource\Item;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+class Family extends Controller
```
Comment (greydnls) [inline src/Http/Controllers/Family.php:11]: I've preferred 1:1 mapping between endpoints and controllers in the past, which allows us to make Controllers invokable. 

```
class ShowFamily
{
    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, array $vars) : ResponseInterface
    {

    }
}
```

What do you think of that?

Comment (ieatkillerbees) [reply src/Http/Controllers/Family.php:11]: That's my preference as well, I was following from the lumen docs.

File: src/Entities/Transformers/Family.php (hunk 3 above)
Comment (ieatkillerbees) [reply src/Entities/Transformers/Family.php:16]: :+1:

**Pull request closed without merging**